include/fmt/chrono.h

```diff
@@ -1550,11 +1588,11 @@ struct chrono_formatter {
     }
   }
 
-  void write(Rep value, int width) {
+  template <typename RepType> void write(RepType value, int width) {
     write_sign();
     if (isnan(value)) return write_nan();
-    uint32_or_64_or_128_t<int> n =
-        to_unsigned(to_nonnegative_int(value, max_value<int>()));
+    uint32_or_64_or_128_t<std::intmax_t> n =
+        to_unsigned(to_nonnegative_int(value, max_value<std::intmax_t>()));
```

Comment:
Let's use `long long` instead of `intmax_t` because it's a built-in type and can also handle 18 digits of precision.

Reply:
As you wish, std::chrono::duration types use std::intmax_t internally that's why I went along with it.